src/controllers/image.controller.ts

```diff
@@ -0,0 +1,66 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import { Request, response, Response } from "express";
+import imageService, { ImageService } from "../services/image.service";
+import sharp from "sharp";
+import { matchedData } from "express-validator";
+
+export class ImageController {
+  constructor(private imageService: ImageService) {}
+
+  public async addImage(req: Request, res: Response): Promise<Response> {
+    try {
+      const { _id, isAdmin } = req.user!;
+      const recipeId: string = matchedData(req).id;
```

Comment:
I saw below you are doing it this way: `const { id } = matchedData(req);`. The above is correct, however it would be good to stick to one style if possible.

If you want to assign id as `recipeId`, you can do: `const { id: recipeId } = matchedData(req);`